reana_client/utils.py

```diff
@@ -198,15 +199,29 @@ def _validate_environment(reana_yaml):
         workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
         _validate_serial_workflow_environment(workflow_steps)
     elif workflow_type == "yadage":
-        logging.warning(
-            "The environment validation is not implemented for Yadage workflows yet."
-        )
+        workflow_steps = reana_yaml["workflow"]["specification"]["stages"]
+        _validate_yadage_workflow_environment(workflow_steps)
     elif workflow_type == "cwl":
         logging.warning(
             "The environment validation is not implemented for CWL workflows yet."
         )
 
 
+def _validate_yadage_workflow_environment(workflow_steps):
+    """Validate environments in REANA yadage workflow.
+
+    :param workflow_steps: List of dictionaries which represents different steps involved in workflow.
+    :raises Warning: Warns user if the workflow environment is invalid in yadage workflow steps.
+    """
+    for stage in workflow_steps:
+        environment = stage["scheduler"]["step"]["environment"]
+        image = "{}:{}".format(environment["image"], environment["imagetag"])
+        image_name, image_tag = _validate_image_tag(image)
+        _image_exists(image_name, image_tag)
+        uid, gids = _get_image_uid_gids(image_name, image_tag)
+        _validate_uid_gids(uid, gids, kubernetes_uid=stage.get("kubernetes_uid"))
```

Comment:
`kubernetes_uid` is not placed there, check for instance`reana-demo-atlas-recast` how it's not taking it into account. It should be something like this:

```python
k8s_uid = (
    environment.get("resources")[0].get("kubernetes_uid")
    if environment.get("resources")
    else None
)
```

*(untested)*

Reply:
Tested on `reana-demo-atlas-recast` and `reana-demo-root6-roofit`. Works fine.